### PR TITLE
[SPARK-48964][SQL][DOCS] Fix the discrepancy between implementation, comment and documentation of option `recursive.fields.max.depth` in ProtoBuf connector

### PR DESCRIPTION
**Pull request by wayneguow**
### What changes were proposed in this pull request?

This PR aims to fix the discrepancy between implementation, comment and documentation of option `recursive.fields.max.depth` in ProtoBuf connector

### Why are the changes needed?

Unify code implementation and documentation description.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass GA.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/SchemaConverters.scala [hunk 1]
```diff
@@ -176,16 +176,16 @@ object SchemaConverters extends Logging {
...
-        // A value greater than 10 is not allowed, and if a protobuf record has more depth for
-        // recursive fields than the allowed value, it will be truncated and some fields may be
-        // discarded.
+        // and recursive fields are not permitted. Setting it to 1 drops all recursive fields,
+        // 2 allows it to be recursed once, and 3 allows it to be recursed twice and so on.
+        // A value less than or equal to 0 or greater than 10 is not allowed, and if a protobuf
+        // record has more depth for recursive fields than the allowed value, it will be truncated
+        // and some fields may be discarded.
         // SQL Schema for protob2uf `message Person { string name = 1; Person bff = 2;}`
         // will vary based on the value of "recursive.fields.max.depth".
```
Comment (wayneguow) [inline connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/SchemaConverters.scala:185]: Current code implementation as belows

**Comment by wayneguow** [comment]
cc @eason-yuchen-liu  @rangadi

**Comment by wayneguow** [comment]
also cc @gengliangwang, please take a look when your have time.

**Comment by wayneguow** [comment]
also cc @HyukjinKwon

**Comment by LuciferYang** [comment]
cc @rangadi FYI

File: connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/SchemaConverters.scala (hunk 1 above)
Comment (LuciferYang) [reply connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/SchemaConverters.scala:185]: Have the existing test cases covered 0 or 1 scenario?

Comment (wayneguow) [reply connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/SchemaConverters.scala:185]: Yes, I checked test cases in `ProtobufFunctionsSuite`.

-1 and 0: https://github.com/apache/spark/blob/34a52ad2ce377dd077c4cf70740ba3ab6c18c739/connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufFunctionsSuite.scala#L452

1: 
https://github.com/apache/spark/blob/34a52ad2ce377dd077c4cf70740ba3ab6c18c739/connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufFunctionsSuite.scala#L977

**Review by HyukjinKwon — APPROVED** [review]

File: connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/SchemaConverters.scala (hunk 1 above)
Comment (LuciferYang) [reply connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/utils/SchemaConverters.scala:185]: OK

**Review by LuciferYang — APPROVED** [review]

**Comment by LuciferYang** [comment]
Merge into master. Thanks @wayneguow and @HyukjinKwon

**Pull request closed without merging**